design/volume-policy-label-selector-criteria.md

```diff
@@ -0,0 +1,180 @@
+# Add Label Selector as a criteria for Volume Policy
+
+## Abstract
+Velero’s volume policies currently support several criteria—such as capacity, storage class, and volume source type—for selecting volumes for backup operations. This update enhances that design by adding a label selector condition that uses Kubernetes native LabelSelector syntax to filter volumes based on the labels of their associated PVCs. This enables users to back up volumes more granularly (for example, by environment, application, or region) while preserving backward compatibility.
+## Background 
+PersistentVolumes(PVs) in Kubernetes are often bound to PersistentVolumeClaims(PVCs) that carry labels representing attributes like environment, application, or region. Matching backup policies against these PVC labels—provides greater flexibility and aligns with how many users manage their workloads. This design entails:
+- Adopting the Kubernetes LabelSelector API for expressing selection criteria.
+- Performing a runtime lookup of the PVC (when a PV has a ClaimRef) so that the PVC’s labels are used during policy evaluation.
+
+## Goals
+- Enable users to specify a PVC label selector in the volume policy so that only volumes whose associated PVCs meet the criteria are backed up.
```

Comment:
Do we need to support NotIn flavor here? E.g., all PVCs with a specific label should not be backed up

Reply:
yeah so using `metav1.LabelSelector` means we will inherit all the supported `matchExpressions`

Reply:
Update design to keep it simple and use `map[string]string`